sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/implementation/KeyVaultCredentialPolicy.java

```diff
@@ -165,6 +183,8 @@ public Mono<Boolean> authorizeRequestOnChallenge(HttpPipelineCallContext context
 
                 if (this.challenge == null) {
                     return Mono.just(false);
+                } else if (claims != null) {
+                    CHALLENGE_CACHE.put(authority, this.challenge.setClaims(claims));
```

Comment:
We don't want to cache claims because the client must use them exactly once, in the token request following the 401 response that provided them

Reply:
I'll remove the claims from the cache.